types/validator_set_test.go

```diff
@@ -206,3 +207,63 @@ func BenchmarkValidatorSetCopy(b *testing.B) {
 		vset.Copy()
 	}
 }
+
+func TestFuzzValidatorSetFromBytes(t *testing.T) {
+	// See Issue https://github.com/tendermint/tendermint/issues/722
+	// which was fixed by https://github.com/tendermint/go-wire/pull/38
+	vectors := [][]byte{
+		{
+			0x01, 0x01, 0x30, 0x01, 0x06, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30,
```

Comment:
yeh not thrilled about this kind of thing in unit tests. way too difficult to maintain.

Perhaps we could start an external testing system that can keep binary encoded things and make sure they deserialize properly for sake of cross implementation testing but this feels to excessive/dirty in the go code.

Reply:
@ebuchman that's malicious code that would be crafted by an attacker, it was auto-generated by the fuzzer hence doesn't directly translate to normal code and the test is testing FromBytes.

Would you like me instead to put these into some directory such as `testdata` instead?

Reply:
And for references for what generated them, it was from https://github.com/tendermint/tendermint/commit/2ba16561d00b6675ee996bd92cad734fae29a318.

Reply:
Maybe what we should do instead is to run this fuzzier once in a while (before every release) to make sure we didn't introduce new bugs?

Reply:
Do we even use the FromBytes function anywhere?! Let's just get rid of it :P

Reply:
Ok well, maybe that's not great, since we obviously deserialize from bytes elsewhere, but maybe we can just use the fuzzers on those, since from what I can tell we don't ever call FromBytes here. Though maybe it's still useful for finding a possible bug like this? 

Also, isn't all this just because we're not using a limit in ReadBinary ?

Reply:
lets run fuzzers on release; https://github.com/tendermint/tendermint/issues/752

Reply:
`Maybe what we should do instead is to run this fuzzier once in a while (before every release) to make sure we didn't introduce new bugs?`
Great and interesting idea, but the problem is that hopefully we remember to do this, or perhaps we can put some test directives to the release automation.

`Also, isn't all this just because we're not using a limit in ReadBinary ?`
Yes, a combination of `0` and malicious vectors, of which 0 as a limit is used allover the place manifests in this bug.

`Do we even use the FromBytes function anywhere?! Let's just get rid of it :P`
Sure, we can get rid of it but at least lets update the go-wire dependency since many places use a limit of 0, unless am mistaken.

Reply:
> perhaps we can put some test directives to the release automation

Yeh we should look to add this.

> at least lets update the go-wire dependency since many places use a limit of 0, 

What's the proposed update? To prevent using 0? Sometimes it's nice like in tests. But maybe it's worth forbidding it from being 0 and just put some effort into the tests to provide a limit